### ISSUE TITLE
typescript; no-useless-constructor fires incorrectly when ctor has ''scoped" arguments

### ISSUE THREAD

**Issue by plastikfan** [eslint/eslint#12964]
**Tell us about your environment**

Node version: v12.13.0
npm version: v6.14.0
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
'@typescript-eslint/parser'

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    node: true,
    browser: true,
    es6: true,
    mocha: true
  },
  extends: [
    'semistandard'
  ],
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly'
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2018, 
    sourceType: 'module'
  },
  plugins: [
    '@typescript-eslint'
  ],
  rules: {
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export class ExpressionBuilderImpl {

  constructor (private converter: jaxom.IConverter, private specSvc: jaxom.ISpecService,
    private parseInfo: jaxom.IParseInfo, private xpath: types.ISelectors) { }

```
As you can see, the constructor implementation is empty, but the arguments are "scoped" (I'm not sure if this is the right terminology); ie they are decorated by 'private' which as you'll know means that they are also declared as member variables, so this is a valid constructor, not a 'useless' one.

```bash
npx eslint -c .eslintrc.js "./lib/**/*.{js,ts,tsx}" "./tests/**/*.{js,ts,tsx}"
```

**What did you expect to happen?**
Rule not to fire.

**What actually happened? Please include the actual, raw output from ESLint.**

λ ~/dev/github/js/zenobia-ts/ feature/eslint* npx eslint -c .eslintrc.js ./lib/regex/expression-builder.impl.ts

/Users/Plastikfan/dev/github/js/zenobia-ts/lib/regex/expression-builder.impl.ts
  13:3  error  Useless constructor  no-useless-constructor

✖ 1 problem (1 error, 0 warnings)

**Are you willing to submit a pull request to fix this bug?**
Willing to assist with investigation, but I don't have the expertise to perform maintenance on eslint.

**Comment by mdjermanovic** [comment]
Similar to #12963, you should probably use [@typescript-eslint/no-useless-constructor](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md).

These [Extension Rules](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules) are extending core rules in order to fix issues you are encountering.

**Comment by plastikfan** [comment]
Ah yes. I'll close this issue and hopefully I can use "typescript-eslint/eslint-plugin" to fix my issues, thanks.
